public/app/plugins/datasource/graphite/migrations.ts

```diff
@@ -0,0 +1,38 @@
+type LegacyAnnotation = {
+  target?: string;
+  tags?: string;
+};
+
+// this becomes the target in the migrated annotations
+const migrateLegacyAnnotation = (json: LegacyAnnotation) => {
+  // return the target annotation
+  if (typeof json.target === 'string' && json.target) {
+    return {
+      fromAnnotations: true,
+      target: json.target,
+      textEditor: true,
+    };
+  }
+
+  // return the tags annotation
+  return {
+    queryType: 'tags',
+    tags: (json.tags || '').split(' '),
+    fromAnnotations: true,
+  };
+};
+
+export const prepareAnnotation = (json: any) => {
```

Comment:
See: https://github.com/grafana/grafana/pull/52546#discussion_r925835623

Reply:
This function handles both old annotations and up to date queries, so I think it is a good case for using the 'any' type. I can add ```// eslint-ignore-next-line``` to this spot though.